quinn-proto/src/streams.rs

```diff
@@ -255,9 +256,15 @@ impl Streams {
     ///
     /// Does not cause the actual RESET_STREAM frame to be sent, just updates internal
     /// state.
-    pub fn reset(&mut self, id: StreamId, stop_reason: Option<VarInt>) -> Option<ResetStatus> {
-        let stream = self.send.get_mut(&id)?;
-        stream.reset(stop_reason)
+    pub fn reset(
+        &mut self,
+        id: StreamId,
+        stop_reason: Option<VarInt>,
+    ) -> (u64, Option<ResetStatus>) {
```

Comment:
Looks like this is flipflopping on the return type within this PR. Is that necessary/useful?

Reply:
The modified return type is needed to update the send window until send window bookkeeping is moved into `Streams` in turn. I could reorder the changes so that reset isn't moved in until the send window is, but that'd be a moderately painful/error-prone rebase, so my inclination is to leave things as-is unless you feel strongly.

Reply:
Fair enough.